clang/tools/c-index-test/core_main.cpp

```diff
@@ -990,6 +992,39 @@ static int replayCachedJob(ArrayRef<const char *> Args,
   return 0;
 }
 
+static int pruneCAS(uint64_t Limit, CXCASDatabases DBs) {
+  CXError Err = nullptr;
+  int64_t Size = clang_experimental_cas_Databases_get_storage_size(DBs, &Err);
+  if (Size == -2) {
+    llvm::errs() << clang_Error_getDescription(Err) << "\n";
+    clang_Error_dispose(Err);
+    return 1;
+  }
+  if (Size == -1) {
+    llvm::errs()
+        << "unsupported clang_experimental_cas_Databases_get_storage_size";
+    return 1;
+  }
+  if (Size == 0) {
+    llvm::errs()
+        << "clang_experimental_cas_Databases_get_storage_size returned 0";
+    return 1;
+  }
+
+  if (clang_experimental_cas_Databases_set_size_limit(DBs, Limit, &Err)) {
+    llvm::errs() << clang_Error_getDescription(Err) << "\n";
```

Comment:
Suggest we add some hint that this error came from `set_size_limit` to differentiate it, same for the below.

Reply:
See https://github.com/apple/llvm-project/pull/7987/commits/87b4dc3a8eb90cd772b979ef7c03156da986bddd